lib/ansible/plugins/terminal/eric_eccli.py

```diff
@@ -0,0 +1,104 @@
+#
+# Copyright (c) 2019 Ericsson AB.
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import json
+import re
+
+from ansible import constants as C
+from ansible.errors import AnsibleConnectionFailure
+from ansible.module_utils._text import to_text, to_bytes
+from ansible.plugins.terminal import TerminalBase
+from ansible.utils.display import Display
+from ansible.module_utils.six import PY3
+
+display = Display()
+
+
+def load_additional_regular_setting(all_eres, all_pres):
+    new_eres = []
+    new_pres = []
+    mode = 0
+    file = None
+    if PY3:
+        python_version = 3
+    else:
+        python_version = 2
+
+    try:
+        config_file = C.ERIC_ECCLI_ADDITIONAL_RE_FILE
```

Comment:
Why do you need to read the terminal regex from a file? Reading the regex from the file doesn't seem right to me.

Reply:
because every device may has different prompt of  terminal command, even though  one   maybe change as version update. We can catch various terminal prompt form device if write terminal regex in additional file,it is convenient to users who couldn't modify code directly. 
what do you think? Thanks.

Reply:
I am working on a feature that allows terminal plugins to be configurable. This allows setting the terminal options like output and error regex per-host basis in host/group vars
For eg:
```
$ls
ios_show_version.yaml     inventory    host_vars
$cat inventory
[ios]
ios01 ansible_host=<hostname> ansible_user=<user> ansible_ssh_pass=<password> ansible_network_os=ios ansible_port=22 ansible_connection=network_cli
$cat host_vars/ios01.yaml
---
ansible_terminal_stderr_re:
  - pattern: '% ?Test error regex'
    flags: 're.I'
ansible_terminal_stdout_re
  - pattern: '[\r\n]?[\w\+\-\.:\/\[\]]+(?:\([^\)]+\)){0,3}(?:[>#]) ?$'

$cat ios_show_version.yaml 
---
- hosts: ios
  gather_facts: no
  tasks:
  - name: get version information
    ios_command:
      commands:
        - show version
```
Or per command as task vars:
```
  - name: get interface
    ios_command:
      commands:
        - show interface
    register: result
    vars:
      ansible_terminal_stderr_re:
        - pattern: '% ?Test error regex'
```

With this feature, you don't require file handling in the terminal plugin, you will need to mention only the configurable option in the plugin documentation. So, for now, you can remove the file handling logic from your terminal plugin.

Reply:
Thank you for detailed examples, but I can't understand why to write prompt regex  in host_vars/ios01.yaml , could you tell me about how  to use it with ansible-playbook command ?  Thank you very much.

Reply:
> why to write prompt regex in host_vars/ios01.yaml 
As you mentioned the prompt regex may vary based on the os version running on the host, mentioning it in host vars provides a convenient way to set the regex for each host. You can also it as group vars based on the inventory.

After the prompt regex are defined as host/group vars Ansible will load it internally when playbook run is triggered.

Reply:
Thanks. Maybe our methods are similar, all need to load regex from a file, so I suspect you have also written some codes to open and read regex from ios01.yaml  like I need to load in additional,but only our inventory path are different.

Reply:
Variable loading is handled by Ansible engine and not specific to any module. Terminal configurable options will work for all the platforms that have a terminal plugin.

Reply:
Thank you for patient explanation. what you mean is that we only write variable regex somewhere, and Ansible will load it automatically, rather than requiring us to add the loading function in own module.
As you mentioned above,after the prompt regex are defined as host/group/task vars Ansible will load it internally when playbook run is triggered. How to link host_vars/ios01.yaml when playbook runs? could you tell me what is the specific command and whether like "ansible-playbook --inventory inventory test_playbook.yaml -vvvv"?

Reply:
> Thank you for patient explanation. what you mean is that we only write variable regex somewhere, and Ansible will load it automatically, rather than requiring us to add the loading function in own module?

That's right


>  How to link host_vars/ios01.yaml when playbook runs? could you tell me what is the specific command and whether like "ansible-playbook --inventory inventory test_playbook.yaml -vvvv"?

For the example, I shared the name of the file is kept the same as the host entry inventory file for Ansible to identify which variables to load for the given host.
For more detials refer Ansible documentation https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html

Reply:
@ganeshrn Thanks,I attempted to move my regex from terminal plugins to playbook file, the display result is hung up when run "ansible-playbook --inventory inventory playbook_eric_eccli_examples.yml -vvvv --tags=examples" command.
cat playbook_eric_eccli_examples.yml:
name: Run a command through eric_eccli module
hosts: confdremote1
connection: network_cli
gather_facts: False
tasks:
#----------------------------------------------
block: 
name: Show version information
eric_eccli_command:
commands: show version
register: result
vars:
ansible_terminal_stdout_re: 
pattern: '[\r\n]?[.][a-zA-Z0-9_.-][>#] ?$'
tags:
examples
cat inventory
[confdremote1]
127.0.0.1 ansible_host=127.0.0.1 ansible_user=test ansible_ssh_pass=test ansible_network_os=eric_eccli ansible_port=8088 ansible_connection=network_cli
but if keep prompt regex in code module is right, where is question? is error about my write syntax?


Resolve conversation

Reply:
@ganeshrn 
could you help me solve above condition? or is something forgot to register? Thanks.

Reply:
As mentioned in this https://github.com/ansible/ansible/pull/59277#discussion_r307589434 the work to make terminal plugins configurable is in progress and I will raise a PR to support it soon. File handling in the terminal plugin to load regex is not the right way as mentioned in the earlier comment.

Reply:
ok, I understand your mean, thank you for the hard work. waiting for good news. I will try  your method after you merge. I have deleted this specific processing  from my module.